packages/react-composites/src/composites/ChatComposite/adapter/OnFetchProfileCallback.ts

```diff
@@ -31,6 +32,68 @@ export type ChatProfile = {
   displayName?: string;
 };
 
+/**
+ * Handle Teams users who don't have display names in participant list
+ * but have them available in chat messages
+ * @private
+ */
+const handleTeamsUsersDisplayNames = async (
+  participants: { [id: string]: ChatParticipant } | undefined,
+  chatMessages: { [id: string]: ChatMessageWithStatus } | undefined,
+  cachedDisplayName: { [id: string]: string },
+  onFetchProfile: OnFetchChatProfileCallback,
+  markForUpdate: () => void
+): Promise<void> => {
+  if (!participants || !chatMessages) {
+    return;
+  }
+
+  // Find participants who are Teams users or have missing/invalid display names
+  for (const [participantId, participant] of Object.entries(participants)) {
+    // Use the utility function to identify Teams users
+    const isTeamsUser = _isIdentityMicrosoftTeamsUser(participantId);
+    const needsDisplayName =
+      !participant.displayName ||
+      participant.displayName.trim() === '' ||
+      participant.displayName === 'undefined' ||
+      // Teams users often don't have display names in participant list
+      // but have them in chat messages
+      isTeamsUser ||
+      !cachedDisplayName[participantId] ||
+      cachedDisplayName[participantId].includes('undefined');
+
+    if (needsDisplayName) {
+      // Look for this participant's display name in chat messages
+      for (const message of Object.values(chatMessages)) {
+        if (message?.sender) {
+          const senderId =
+            message.sender.kind === 'microsoftTeamsUser' && message.sender.rawId
+              ? message.sender.rawId
+              : message.sender.kind === 'communicationUser' && message.sender.communicationUserId
+                ? message.sender.communicationUserId
+                : message.sender.kind === 'phoneNumber' && message.sender.phoneNumber
+                  ? message.sender.phoneNumber
+                  : message.sender.kind === 'microsoftTeamsApp' && message.sender.rawId
+                    ? message.sender.rawId
+                    : message.sender.kind === 'unknown' && message.sender.id
+                      ? message.sender.id
+                      : null;
+
+          if (senderId === participantId && message.senderDisplayName && message.senderDisplayName.trim() !== '') {
+            // Found a display name for this participant in a chat message
+            const profile = await onFetchProfile(participantId, { displayName: message.senderDisplayName });
+            if (profile?.displayName && cachedDisplayName[participantId] !== profile?.displayName) {
+              cachedDisplayName[participantId] = profile?.displayName;
+              markForUpdate();
+              break; // Found display name, no need to check more messages for this participant
+            }
+          }
+        }
+      }
+    }
+  }
+};
```

Comment:
nit: As an optimization, instead of a for loop in a for loop, we can make a map with senderId as keys and display name and displayName as values and then loop through each participant by checking the map for a display name. 
Right now it is O(P x M) were P is the number of participants and M is the number of message and with this optimization it would be O(P + M).